### PR TITLE
[confignet] Mark as stable

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:**
Mark `confignet` as Stable

**Link to tracking Issue:** 
Closes https://github.com/open-telemetry/opentelemetry-collector/issues/9045

### CODE REVIEW

**Review by atoulme — APPROVED** [review]
LGTM! I have reviewed the code and the test coverage, everything looks great.

**Review by dmitryax — APPROVED** [review]
I've been thinking how to reduce code duplication between `TCPAddrConfig`  and `AddrConfig`, but cannot see a clean simple way. Go isn't very flexible sometimes.

File: versions.yaml
Comment (astencel-sumo) [inline versions.yaml:12]: It's funny how this module goes from 0.x not to 1.0 but straight to 1.3 :thinking:

File: .chloggen/confignet-mark-stable.yaml
Comment (astencel-sumo) [inline .chloggen/confignet-mark-stable.yaml:25]:
```suggestion
change_logs: [api]
```

File: versions.yaml
Comment (TylerHelmuth) [reply versions.yaml:12]: `1.0`, `1.1`, and `1.2` are overrated lol

File: .chloggen/confignet-mark-stable.yaml
Comment (TylerHelmuth) [reply .chloggen/confignet-mark-stable.yaml:25]: This is technically an API change, but we've been putting these releases in the user-facing config bc its more visible and an important milestone.

**Review by astencel-sumo — APPROVED** [review]

**Review by mx-psi — COMMENTED** [review]
By stabilizing this, we commit to two things from confmap:

1. Supporting `mapstructure` YAML tags. I am fine with that, I don't think there is much of a point on changing it, but I know there has been some discussion about this (@atoulme may have more details?). Is this resolved?
2. Supporting `UnmarshalText` on config unmarshaling. This is a standard library interface, so I think it is a pretty safe assumption.

We also commit to one thing from `component`:
1. `AddrConfig` implements `component.ConfigValidator`: https://github.com/open-telemetry/opentelemetry-collector/blob/15201f1e5967fd1bc1e51c118178ab109170fe07/config/confignet/confignet.go#L106. We are commiting to keep on supporting that interface.

For the implicit dependency on `component`, I don't really get why we are adding `Validate()`: the `UnmarshalText` method from the transport should already do validation at unmarshaling time. Shouldn't that be enough? If we can remove that method, we eliminate one implicit dependency.

**Comment by TylerHelmuth** [comment]
>  I don't really get why we are adding Validate()

@mx-psi I believe `Validate()` should be included since `AddrConfig.Transport` is a public field.  Users are able to set `AddrConfig.Transport` manually, bypassing `TransportType.UnmarshalText`. See https://github.com/open-telemetry/opentelemetry-collector/pull/9385#discussion_r1477104190.

**Comment by TylerHelmuth** [comment]
@mx-psi based on your comments it sounds like we have sufficient dependencies on `component` and `confmap` to hold off marking this stable until those 2 modules are stable.

**Comment by mx-psi** [comment]
> @mx-psi based on your comments it sounds like we have sufficient dependencies on `component` and `confmap` to hold off marking this stable until those 2 modules are stable.

Or at least I think if we merge this, we should have a way to validate that we are not changing these assumptions

**Comment by TylerHelmuth** [comment]
Will reopen once `component` and `confmap` are stable

**Comment by atoulme** [comment]
There is no dependency to component left and confmap is stable, reopening.

**Review by mx-psi — APPROVED** [review]
I think it's pretty clear that `Validate` is here to stay, which was the only dependency left with component. I vote we merge this.

@open-telemetry/collector-approvers I will merge this after v0.109.0 is released unless somebody blocks the PR

**Review by songy23 — APPROVED** [review]

**Review by MovieStoreGuy — APPROVED** [review]

**Pull request merged**
